### PR TITLE
cdc: update description about partition dispatcher and add column selector descriptions.

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: ticdc/ticdc-sink-to-kafka.md [hunk 1]
```diff
@@ -224,12 +224,33 @@ Topic 表达式的基本规则为 `[prefix][{schema}][middle][{table}][suffix]`
...
+]
+```
+
+- 任何属于库 `test` 的表，均使用 `index-value` 分发规则，即使用主键或者唯一索引的值计算 Partition 编号。(有主键则使用主键，无主键使用最短的唯一索引)
+- 任何属于库 `test1` 的表，均使用 `index-value` 分发规则，并且使用名为 `index1` 的索引的所有列的值计算 Partition 编号，如果指定的索引不存在，则报错。(`index-name` 指定的索引必须是唯一索引)
+- 任何属于库 `test2` 的表，均使用 `columns` 分发规则，并且使用列 `id` 和 `a` 的值计算 Partition 编号。如果任一列不存在，则报错。
+- 任何属于库 `test3` 的表，均使用 `table` 分发规则。
+- 对于属于库 `test4` 的表，因为不匹配上述任何一个规则，所以使用默认的 `default`，即 `table` 分发规则。
+
+如果一张表，匹配了多个分发规则，以第一个匹配的规则为准。
```
Comment (Oreoxmt) [inline ticdc/ticdc-sink-to-kafka.md:253]:
````suggestion
- default：默认使用 table 分发规则。使用所属库名和表名计算 partition 编号，一张表的数据被发送到相同的 partition。单表数据只存在于一个 partition 中并保证有序，但是发送吞吐量有限，无法通过添加消费者的方式提升消费速度。
- index-value：使用事件所属表的主键、唯一索引或者明确指定名字的索引值计算 partition 编号，一张表的数据被发送到多个 partition。单表数据被发送到多个 partition 中，每个 partition 中的数据有序，可以通过添加消费者的方式提升消费速度。
- columns：使用明确指定的列值计算 partition 编号。一张表的数据被发送到多个 partition。单表数据被发送到多个 partition 中，每个 partition 中的数据有序，可以通过添加消费者的方式提升消费速度。
- table：使用事件所属的表的库名和表名计算 partition 编号。
- ts：使用事件的 commitTs 计算 partition 编号。一张表的数据被发送到多个 partition。单表数据被发送到多个 partition 中，每个 partition 中的数据有序，可以通过添加消费者的方式提升消费速度。一条数据的多次修改可能被发送到不同的 partition 中。消费者消费进度不同可能导致消费端数据不一致。因此，消费端需要对来自多个 partition 的数据按 commitTs 排序后再进行消费。

以如下示例配置文件中的 `dispatchers` 配置项为例：

```toml
[sink]
dispatchers = [
    {matcher = ['test.*'], partition = "index-value"},
    {matcher = ['test1.*'], partition = "index-value", index-name = "index1"},
    {matcher = ['test2.*'], partition = "columns", columns = ["id", "a"]},
    {matcher = ['test3.*'], partition = "table"},
]
```

- 任何属于库 `test` 的表均使用 `index-value` 分发规则，即使用主键或者唯一索引的值计算 partition 编号。如果有主键则使用主键，否则使用最短的唯一索引。
- 任何属于库 `test1` 的表均使用 `index-value` 分发规则，并且使用名为 `index1` 的索引的所有列的值计算 partition 编号。如果指定的索引不存在，则报错。注意，`index-name` 指定的索引必须是唯一索引。
- 任何属于库 `test2` 的表均使用 `columns` 分发规则，并且使用列 `id` 和 `a` 的值计算 partition 编号。如果任一列不存在，则报错。
- 任何属于库 `test3` 的表均使用 `table` 分发规则。
- 对于属于库 `test4` 的表，因为不匹配上述任何一个规则，所以使用默认的 `default`，即 `table` 分发规则。

如果一张表，匹配了多个分发规则，以第一个匹配的规则为准。
````

File: ticdc/ticdc-sink-to-kafka.md [hunk 2]
```diff
@@ -238,15 +259,41 @@ partition 分发器用 partition = "xxx" 来指定，支持 default、ts、index
...
+
+- 对于表 `test.t1`，只发送 `a` 和 `b` 两列的数据。
+- 任何属于库 `test` 的除了 t1 的表，发送除 `b` 列之外的所有列的数据。
+- 对于表 `test1.t1`，发送所有以 `column` 开头的列，但是不发送 `column1` 列的数据。
+- 对于表 `test3.t`，发送所有以 `column` 开头且长度为 6 的列，但是不发送 `column1` 列的数据。
+- 对于不匹配任意一个规则的表，不做列过滤处理，发送所有列。
+
+> **注意：**
+> 如果一个列是主键，或者唯一索引的一部分，它必须被同步，无法被过滤。在创建 changefeed 或 changefeed 运行时，如果发现该原则被违背，changefeed 会报错。
 > ```
```
Comment (Oreoxmt) [inline ticdc/ticdc-sink-to-kafka.md:297]:
```suggestion
- 对于表 `test.t1`，只发送 `a` 和 `b` 两列的数据。
- 对于属于库 `test` 的表（除 `t1` 外），发送除 `b` 列之外的所有列的数据。
- 对于表 `test1.t1`，发送所有以 `column` 开头的列，但是不发送 `column1` 列的数据。
- 对于表 `test3.t`，发送所有以 `column` 开头且列名长度为 6 的列，但是不发送 `column1` 列的数据。
- 不匹配任何规则的表将不进行列过滤，发送所有列的数据。

> **注意：**
>
> 如果某列是主键或唯一索引的一部分，则该列必须被包含在同步过程中，无法被过滤。在创建 changefeed 或 changefeed 运行时，如果发现此规则被违背，changefeed 会报错。
```

File: ticdc/ticdc-sink-to-kafka.md
Comment (Oreoxmt) [inline ticdc/ticdc-sink-to-kafka.md:1]: Please update the `ticdc-changefeed-config.md` file:
https://github.com/pingcap/docs-cn/blob/7204c909d92c75f2f398c4d2518abe891c5923f6/ticdc/ticdc-changefeed-config.md?plain=1#L111-L123

File: ticdc/ticdc-changefeed-config.md [hunk 3]
```diff
@@ -116,10 +116,19 @@ enable-table-across-nodes = false
...
-#    {matcher = ['test3.*', 'test4.*'], topic = "Topic 表达式 2", partition = "index-value" },
+#    {matcher = ['test1.*', 'test2.*'], topic = "Topic 表达式 1", partition = "index-value"},
+#    {matcher = ['test3.*', 'test4.*'], topic = "Topic 表达式 2", partition = "index-value", index-name="index1"},
 #    {matcher = ['test1.*', 'test5.*'], topic = "Topic 表达式 3", partition = "table"},
-#    {matcher = ['test6.*'], partition = "ts"}
+#    {matcher = ['test6.*'], partition = "columns", columns = "['a', 'b']"}
+#    {matcher = ['test7.*'], partition = "ts"}
+# ]
+
+# 仅对 Kafka Sink 生效，column-selectors 用于选择部分列进行同步。
```
Comment (Oreoxmt) [inline ticdc/ticdc-changefeed-config.md:126]:
```suggestion
# column-selectors 从 v7.5.0 开始引入，仅对 Kafka Sink 生效。
# column-selectors 用于选择部分列进行同步。
```

File: ticdc/ticdc-changefeed-config.md [hunk 4]
```diff
@@ -116,10 +116,20 @@ enable-table-across-nodes = false
 # 注意：该参数只有当下游为消息队列时，才会生效。
 # 注意：当下游 MQ 为 Pulsar 时，如果 partition 的路由规则未指定为 'ts', 'index-value', 'table', 'default' 中的任意一个，那么将会使用你设置的字符串作为每一条 Pulsar message 的 key 进行路由。例如，如果你指定的路由规则为 'code' 字符串，那么符合该 matcher 的所有 Pulsar message 都将会以 'code' 作为 key 进行路由。
 # dispatchers = [
-#    {matcher = ['test1.*', 'test2.*'], topic = "Topic 表达式 1", partition = "ts" },
-#    {matcher = ['test3.*', 'test4.*'], topic = "Topic 表达式 2", partition = "index-value" },
+#    {matcher = ['test1.*', 'test2.*'], topic = "Topic 表达式 1", partition = "index-value"},
```
Comment (nongfushanquan) [inline ticdc/ticdc-changefeed-config.md:119]: 这两者之间的区别要说明一下，不然用户不会理解区别。

File: ticdc/ticdc-changefeed-config.md [hunk 5]
```diff
@@ -116,10 +116,20 @@ enable-table-across-nodes = false
...
+#    {matcher = ['test3.*', 'test4.*'], topic = "Topic 表达式 2", partition = "index-value", index-name="index1"},
 #    {matcher = ['test1.*', 'test5.*'], topic = "Topic 表达式 3", partition = "table"},
-#    {matcher = ['test6.*'], partition = "ts"}
+#    {matcher = ['test6.*'], partition = "columns", columns = "['a', 'b']"}
+#    {matcher = ['test7.*'], partition = "ts"}
+# ]
+
+# column-selectors 从 v7.5.0 开始引入，仅对 Kafka Sink 生效。
```
Comment (nongfushanquan) [inline ticdc/ticdc-changefeed-config.md:126]: 要说明一下 columns 里面出现的列的限制

File: ticdc/ticdc-changefeed-config.md (hunk 4 above)
Comment (3AceShowHand) [reply ticdc/ticdc-changefeed-config.md:119]: 这一块儿主要是配置内容，主要目的是给个示例。

后面有对配置项更加详细的说明。

File: ticdc/ticdc-changefeed-config.md (hunk 5 above)
Comment (3AceShowHand) [reply ticdc/ticdc-changefeed-config.md:126]: 列的限制相关的说明，在 column-selectors 功能说明里面有。

File: ticdc/ticdc-sink-to-kafka.md [hunk 6]
```diff
@@ -238,17 +259,43 @@ partition 分发器用 partition = "xxx" 来指定，支持 default、ts、index
...
+
+- 对于表 `test.t1`，只发送 `a` 和 `b` 两列的数据。
+- 对于属于库 `test` 的表（除 `t1` 外），发送除 `b` 列之外的所有列的数据。
+- 对于表 `test1.t1`，发送所有以 `column` 开头的列，但是不发送 `column1` 列的数据。
+- 对于表 `test3.t`，发送所有以 `column` 开头且列名长度为 6 的列，但是不发送 `column1` 列的数据。
+- 不匹配任何规则的表将不进行列过滤，发送所有列的数据。
+
+> **注意：**
+>
+> 一张表存在主键时，主键列必须被包含在同步过程中。不含有主键时，唯一索引列必须被包含在同步过程中。在创建 changefeed 或 changefeed 运行时，如果发现此规则被违背，changefeed 会报错。
```
Comment (nongfushanquan) [inline ticdc/ticdc-sink-to-kafka.md:297]: 经过 column selector 后，表中的数据必须要有主键或者唯一键被同步，否则在 changefeed  创建或运行时，会报错。

**Review by nongfushanquan — APPROVED** [review]

File: ticdc/ticdc-sink-to-kafka.md [hunk 7]
```diff
@@ -238,17 +259,43 @@ partition 分发器用 partition = "xxx" 来指定，支持 default、ts、index
...
+
+- 对于表 `test.t1`，只发送 `a` 和 `b` 两列的数据。
+- 对于属于库 `test` 的表（除 `t1` 外），发送除 `b` 列之外的所有列的数据。
+- 对于表 `test1.t1`，发送所有以 `column` 开头的列，但是不发送 `column1` 列的数据。
+- 对于表 `test3.t`，发送所有以 `column` 开头且列名长度为 6 的列，但是不发送 `column1` 列的数据。
+- 不匹配任何规则的表将不进行列过滤，发送所有列的数据。
+
+> **注意：**
+>
+> 经过 column selector 规则过滤后，表中的数据必须要有主键或者唯一键被同步，否则在 changefeed 创建或运行时会报错。
```
Comment (Oreoxmt) [inline ticdc/ticdc-sink-to-kafka.md:297]:
```suggestion
> 经过 `column-selectors` 规则过滤后，表中的数据必须要有主键或者唯一键被同步，否则在 changefeed 创建或运行时会报错。
```

**Review by Oreoxmt — APPROVED** [review]

File: ticdc/ticdc-sink-to-kafka.md [hunk 8]
```diff
@@ -224,12 +224,33 @@ Topic 表达式的基本规则为 `[prefix][{schema}][middle][{table}][suffix]`
 
 ### Partition 分发器
 
-partition 分发器用 partition = "xxx" 来指定，支持 default、ts、index-value、table 四种 partition 分发器，分发规则如下：
+partition 分发器用 `partition = "xxx"` 来指定，支持 default、index-value、columns、table 和 ts 共五种 partition 分发器，分发规则如下：
```
Comment (ran-huang) [inline ticdc/ticdc-sink-to-kafka.md:227]:
```suggestion
partition 分发器用 `partition = "xxx"` 来指定，支持 `default`、`index-value`、`columns`、`table` 和 `ts` 共五种 partition 分发器，分发规则如下：
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 9]
```diff
@@ -224,12 +224,33 @@ Topic 表达式的基本规则为 `[prefix][{schema}][middle][{table}][suffix]`
...
 
-- default：按照 table 分发
-- ts：以行变更的 commitTs 做 Hash 计算并进行 event 分发
-- index-value：以表的主键或者唯一索引的值做 Hash 计算并进行 event 分发
-- table：以表的 schema 名和 table 名做 Hash 计算并进行 event 分发
+- default：默认使用 table 分发规则。使用所属库名和表名计算 partition 编号，一张表的数据被发送到相同的 partition。单表数据只存在于一个 partition 中并保证有序，但是发送吞吐量有限，无法通过添加消费者的方式提升消费速度。
+- index-value：使用事件所属表的主键、唯一索引或由 `index-name` 指定的索引的值计算 partition 编号，一张表的数据被发送到多个 partition。单表数据被发送到多个 partition 中，每个 partition 中的数据有序，可以通过添加消费者的方式提升消费速度。
+- columns：使用由 `columns` 指定的列的值计算 partition 编号。一张表的数据被发送到多个 partition。单表数据被发送到多个 partition 中，每个 partition 中的数据有序，可以通过添加消费者的方式提升消费速度。
+- table：使用事件所属的表的库名和表名计算 partition 编号。
+- ts：使用事件的 commitTs 计算 partition 编号。一张表的数据被发送到多个 partition。单表数据被发送到多个 partition 中，每个 partition 中的数据有序，可以通过添加消费者的方式提升消费速度。一条数据的多次修改可能被发送到不同的 partition 中。消费者消费进度不同可能导致消费端数据不一致。因此，消费端需要对来自多个 partition 的数据按 commitTs 排序后再进行消费。
```
Comment (ran-huang) [inline ticdc/ticdc-sink-to-kafka.md:233]:
```suggestion
- `default`：默认使用 table 分发规则。使用所属库名和表名计算 partition 编号，一张表的数据被发送到相同的 partition。单表数据只存在于一个 partition 中并保证有序，但是发送吞吐量有限，无法通过添加消费者的方式提升消费速度。
- `index-value`：使用事件所属表的主键、唯一索引或由 `index-name` 指定的索引的值计算 partition 编号，一张表的数据被发送到多个 partition。单表数据被发送到多个 partition 中，每个 partition 中的数据有序，可以通过添加消费者的方式提升消费速度。
- `columns`：使用由 `columns` 指定的列的值计算 partition 编号。一张表的数据被发送到多个 partition。单表数据被发送到多个 partition 中，每个 partition 中的数据有序，可以通过添加消费者的方式提升消费速度。
- `table`：使用事件所属的表的库名和表名计算 partition 编号。
- `ts`：使用事件的 commitTs 计算 partition 编号。一张表的数据被发送到多个 partition。单表数据被发送到多个 partition 中，每个 partition 中的数据有序，可以通过添加消费者的方式提升消费速度。一条数据的多次修改可能被发送到不同的 partition 中。消费者消费进度不同可能导致消费端数据不一致。因此，消费端需要对来自多个 partition 的数据按 commitTs 排序后再进行消费。
```

File: ticdc/ticdc-sink-to-kafka.md [hunk 10]
```diff
@@ -238,17 +259,43 @@ partition 分发器用 partition = "xxx" 来指定，支持 default、ts、index
...
+    {matcher = ['test.*'], columns = ["*", "!b"]},
+    {matcher = ['test1.t1'], columns = ['column*', '!column1']},
+    {matcher = ['test3.t'], columns = ["column?", "!column1"]},
+]
+```
+
+- 对于表 `test.t1`，只发送 `a` 和 `b` 两列的数据。
+- 对于属于库 `test` 的表（除 `t1` 外），发送除 `b` 列之外的所有列的数据。
+- 对于表 `test1.t1`，发送所有以 `column` 开头的列，但是不发送 `column1` 列的数据。
+- 对于表 `test3.t`，发送所有以 `column` 开头且列名长度为 6 的列，但是不发送 `column1` 列的数据。
```
Comment (ran-huang) [inline ticdc/ticdc-sink-to-kafka.md:292]: 这里是不是写错了，column 已经有 6 个字符
```suggestion
- 对于表 `test3.t`，发送所有以 `column` 开头且列名长度为 7 的列，但是不发送 `column1` 列的数据。
```

**Review by ran-huang — APPROVED** [review]

**Review by ran-huang — APPROVED** [review]

**Pull request merged**